aiconfig-docs/docs/usage-guide.md

```diff
@@ -23,46 +23,138 @@ import constants from '@site/core/tabConstants';
 
 ## Using an `aiconfig`
 
+### Resolve a prompt
+Resolving a prompt will take a Prompt and its settings from the aiconfig and convert it into completion data that can be used to generate a response. This is useful for debugging and testing purposes.
+
+```python
+config.resolve(prompt = "get_activities", params = {} )
+```
+```typescript
+config.resolve("get_activities")
+```
+Sample output:
+```
+{
+ 'top_p': 1,
+ 'temperature': 1,
+ 'model': 'gpt-3.5-turbo',
+ 'messages': [{'content': 'Tell me 10 fun attractions to do in NYC.','role': 'user'}]
+}
+```
+
+### Serializing data into a prompt
+The serialize steps lets you take model Completion Params and convert them To Config Prompts
+
+Take this sample api call to openai:
+```python
+openai.Chat.completions.create(**{
+ 'top_p': 1,
+ 'temperature': 1,
+ 'model': 'gpt-3.5-turbo',
+ 'messages': [{'content': 'Tell me 10 fun attractions to do in NYC.','role': 'user'}]
+})
+```
+We can convert this into a prompt using the serialize method
+```python
+data = {
+ 'top_p': 1,
+ 'temperature': 1,
+ 'model': 'gpt-3.5-turbo',
+ 'messages': [{'content': 'Tell me 10 fun attractions to do in NYC.','role': 'user'}]
+}
+new_prompt = config.serialize("gpt-3.5-turbo",data, prompt_name="get_activities")
+config.add_prompt(new_prompt.name, new_prompt)
+```
+```typescript
+```
+
+
 ### Run a prompt
+You can execute a single prompt using your AIConfig anytime after you've loaded your config into memory. This allows you to obtain a generated response from the model.
+
+```python
+# Example code to run a prompt
+response = aiconfig_runtime.run(prompt_name="my_prompt")
+print(response)
+```
+Sample Output: 
+```
+Execute Result(
+  yada yada yada
+)
+```
+
+AIConfig will automatically store the response in a standardized format, namely `ExecuteResult`
+
+#### Outputs
+
+On Run, a model parser will store the response in a standardized format, namely `ExecuteResult`. This is a dictionary like object with the following attributes:
+- output_type
+- execution_count
+- data
+- mime_type: Optional[str] = None
+- metadata: Dict[str, Any]
```

Comment:
I don't think this is that useful either

Reply:
removed